src/tools/clippy/tests/ui/significant_drop_in_scrutinee.stderr

```diff
@@ -0,0 +1,135 @@
+error: temporary with significant drop in match scrutinee
+  --> $DIR/significant_drop_in_scrutinee.rs:46:11
+   |
+LL |     match mutex.lock().unwrap().foo() { // Should trigger lint
+   |           ^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: `-D clippy::significant-drop-in-scrutinee` implied by `-D warnings`
+help: try getting the temporary outside the match
+   |
+LL ~     let value = mutex.lock().unwrap().foo();
+LL ~     match value { // Should trigger lint
+   |
+
+error: temporary with significant drop in match scrutinee
+  --> $DIR/significant_drop_in_scrutinee.rs:113:11
+   |
+LL |     match s.lock_m().get_the_value() { // Should trigger lint
+   |           ^^^^^^^^^^
+   |
+help: try getting the temporary outside the match
+   |
+LL ~     let value = s.lock_m().get_the_value();
+LL ~     match value { // Should trigger lint
+   |
+
+error: temporary with significant drop in match scrutinee
+  --> $DIR/significant_drop_in_scrutinee.rs:129:11
+   |
+LL |     match s.lock_m_m().get_the_value() { // Should trigger lint
+   |           ^^^^^^^^^^^^
+   |
+help: try getting the temporary outside the match
+   |
+LL ~     let value = s.lock_m_m().get_the_value();
+LL ~     match value { // Should trigger lint
+   |
+
+error: temporary with significant drop in match scrutinee
+  --> $DIR/significant_drop_in_scrutinee.rs:173:11
+   |
+LL |     match counter.temp_increment().len() { // Should trigger lint
+   |           ^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+help: try getting the temporary outside the match
+   |
+LL ~     let value = counter.temp_increment().len();
+LL ~     match value { // Should trigger lint
+   |
+
+error: temporary with significant drop in match scrutinee
+  --> $DIR/significant_drop_in_scrutinee.rs:192:12
+   |
+LL |     match (mutex1.lock().unwrap().s.len(), true) { // Should trigger lint
+   |            ^^^^^^^^^^^^^^^^^^^^^^
+   |
+help: try getting the temporary outside the match
+   |
+LL ~     let value = mutex1.lock().unwrap().s;
+LL ~     match value.len(), true) { // Should trigger lint
```

Comment:
This suggestion is broken. With a multispan suggestion, you should be able to suggest replacing the span of the thing you moved in front of the `match` expr and suggest replacing it with `value` and then a separate suggestion, that just suggests inserting `let value = ...` in front of the `match` expr.

Reply:
I'm not 100% sure how to best do this myself, so I asked on [Zulip](https://rust-lang.zulipchat.com/#narrow/stream/147480-t-compiler.2Fwg-diagnostics/topic/Best.20way.20to.20suggest.20inserting.20code.20before.20expr.20.2394206/near/275347241)

Reply:
Great catch! Thank you!